mirrord/kube/src/api/runtime.rs

```diff
@@ -75,24 +75,53 @@ pub struct RuntimeData {
 }
 
 impl RuntimeData {
+    /// Extracts data needed to create the mirrord-agent targeting the given [`Pod`].
+    /// Verifies that the [`Pod`] is ready to be a target:
+    /// 1. pod is in "Running" phase,
+    /// 2. pod is not in deletion,
+    /// 3. all regular containers are ready (ephemeral containers are not inspected),
```

Comment:
Why do we need all regular containers to be ready and not just a viable target one?
And why don't we care about `initContainerStatuses`?

Reply:
Regarding second question - I think all `ready` containers imply that init containers have finished successfully

Reply:
> init containers run to completion before any app containers start

But not [sidecar containers](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/):
> If an init container is created with its restartPolicy set to Always, it will start and remain running during the entire life of the Pod.